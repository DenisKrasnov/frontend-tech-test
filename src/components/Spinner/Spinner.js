import React from "react";

import styles from "./Spinner.scss";

const Spinner = () => <span className={styles.Spinner} />;

export default Spinner;
