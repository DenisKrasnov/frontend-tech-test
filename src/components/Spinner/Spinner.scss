.Spinner {
  box-sizing: border-box;
  display: inline-block;
  vertical-align: middle;
  width: 22px;
  height: 22px;
  border: 2px solid #fff;
  border-top-color: $button-primary-color;
  border-radius: 50%;
  animation: spinner 1.25s infinite ease-in-out;
}

@keyframes spinner {
  25% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(720deg);
  }
}
