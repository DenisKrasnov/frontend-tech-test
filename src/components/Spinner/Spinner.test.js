import React from "react";
import { shallow } from "enzyme";

import Spinner from "./Spinner";

describe("Spinner", () => {
  it("renders", () => {
    const component = shallow(<Spinner />);

    expect(component).toMatchSnapshot();
  });
});
