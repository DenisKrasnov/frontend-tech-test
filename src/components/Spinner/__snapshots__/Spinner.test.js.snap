// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders 1`] = `
<span
  className="Spinner"
/>
`;
